skbio/io/_base.py

```diff
@@ -185,3 +186,25 @@ def _format_fasta_like_records(generator, id_whitespace_replacement,
                 "scores associated with it." % cardinal_to_ordinal(idx + 1))
 
         yield header, seq.sequence, seq.quality
+
+
+def _line_generator(fh, skip_blanks=False):
+    for line in fh:
+        line = line.strip()
+        if line or not skip_blanks:
+            yield line
+
+
+def _too_many_blanks(fh, max_blanks):
+    count = 0
+    too_many = False
+    for line in _line_generator(fh):
+        if line:
+            break
+        else:
+            count += 1
+            if count > max_blanks:
+                too_many = True
+                break
+    fh.seek(0)
```

Comment:
Correct me if I'm wrong but it seems like using `seek` within the sniffer isn't a problem from the user's perspective because we're using a copy of the file handle, which shouldn't affect the user's copy, right? It appears that another consequence of this that if the user `seek`s the file, the sniffer still starts at the beginning of the file, but the generator will respect the seek. Am I understanding that correctly?